src/Native/System.Native/pal_process.cpp

```diff
@@ -417,3 +427,51 @@ extern "C" char* GetCwd(char* buffer, int32_t bufferSize)
 
     return getcwd(buffer, UnsignedCast(bufferSize));
 }
+
+#if HAVE_SCHED_SETAFFINITY
+static void ConvertPalToNativeCpuSet(const CpuSetBits& pal, cpu_set_t& native)
+{
+    memcpy(native.__bits, pal.Bits, ARRAY_SIZE(native.__bits));
```

Comment:
I'll add a check when sizeof(pal) > sizeof(native) that will look at the bits from the end of pal until the end of native and make sure they are all 0 so that we won't truncate data. If we would truncate, I'll set errno to EINVAL so it will bubble up to the managed layer